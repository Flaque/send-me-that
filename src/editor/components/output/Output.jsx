import React from 'react';
import Frame from 'react-frame-component';

class Output extends React.Component {
  render() {
    const head = `<script>${this.props.code}</script>`;
    return (
      <div className="output">
        <Frame head={<script>{this.props.code}</script>}>
          <div id="console">
          </div>
        </Frame>
      </div>
    )
  }
}

export default Output;
