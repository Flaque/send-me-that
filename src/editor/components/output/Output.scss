#console {
  height: 100px;
  background: #383838;
  border-top: 1px solid #ccc;
  color: white;
  padding: 10px;
  overflow: scroll;

  .console-item {
    width: 100%;
    font-family: "Courier New", Courier;
  }
}
