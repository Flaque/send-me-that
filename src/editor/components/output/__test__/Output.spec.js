import React from 'react';
import Output from '../Output.jsx';
import renderer from 'react-test-renderer';
import ReactTestUtils from 'react-addons-test-utils';

test('Test that Output is still the same', () => {

  // Shallow Renderer here because react-frame-component has strange
  // issues with full snapshots
  const shallowRenderer = ReactTestUtils.createRenderer();
  const result = shallowRenderer.render(<Output />);
  expect(result).toMatchSnapshot();
})
