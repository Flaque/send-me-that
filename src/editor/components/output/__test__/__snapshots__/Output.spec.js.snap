exports[`test Test that Output is still the same 1`] = `
<div
  className="output bordered">
  <div
    id="console" />
</div>
`;
